modules/core/src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -249,15 +249,15 @@ sealed class FundingContributionFailure {
 data class FundingContributions(val inputs: List<InteractiveTxInput.Outgoing>, val outputs: List<InteractiveTxOutput.Outgoing>) {
     companion object {
         /** Compute our local splice contribution using all the funds available in our wallet. */
-        fun computeSpliceContribution(isInitiator: Boolean, commitment: Commitment, walletInputs: List<WalletState.Utxo>, localOutputs: List<TxOut>, targetFeerate: FeeratePerKw): Satoshi {
+        fun computeSpliceContribution(isInitiator: Boolean, commitment: Commitment, walletInputs: List<WalletState.Utxo>, localOutputs: List<TxOut>, liquidityPurchase: Boolean, targetFeerate: FeeratePerKw): Satoshi {
             val weight = computeWeightPaid(isInitiator, commitment, walletInputs, localOutputs)
             val fees = Transactions.weight2fee(targetFeerate, weight)
             return when {
                 // When buying inbound liquidity, we may not have enough funds in our current balance to pay on-chain fees.
                 // The maximum amount we can use for on-chain fees is our current balance, which is fine because:
                 //  - this will simply result in a splice transaction with a lower feerate than expected
                 //  - liquidity fees will be paid later from future HTLCs relayed to us
-                walletInputs.isEmpty() && localOutputs.isEmpty() -> -(fees.min(commitment.localCommit.spec.toLocal.truncateToSatoshi()))
+                walletInputs.isEmpty() && localOutputs.isEmpty() && liquidityPurchase -> -(fees.min(commitment.localCommit.spec.toLocal.truncateToSatoshi()))
```

Comment:
If we're doing a splice out, `localOutputs.isEmpty()` would already be false, so this only changes the behavior for a splice cpfp ?